src/content/developers/docs/smart-contracts/composability/index.md

```diff
@@ -8,7 +8,61 @@ incomplete: true
+
+## What is composability? {#what-is-composability}
+
+Composability is the idea that distinct components can be integrated or combined to create new systems or outputs. In the context of blockchain development, composability means existing pieces of software can be reused to build new applications.
+
+A good way to understand composability is to think of composable elements as Lego blocks. Each Lego is designed to fit into another, such that you can build complex structures by combining different Lego blocks. 
+
+In Ethereum, every smart contract is a Lego of sorts—you can use smart contracts from other projects as building blocks for your project. This means you don't have to spend time reinventing the wheel or building from scratch.
+
+## How does composability work? {#how-does-composability-work}
+
+As we said in the beginning, Ethereum smart contracts are like public APIs. This means anyone can interact with them or integrate them into dapps for added functionality.
+
+Smart contract composability generally works off three principles: modularity, autonomy, and discoverability: 
+
+**1. Modularity**: This is the ability of individual components to perform a specific task. In Ethereum, every smart contract has specific use-case (as shown in the Uniswap example).
+
+**2. Autonomy**: Composable components must be able to operate independently. Each smart contract in Ethereum is self-executing and can function without relying on other parts of the system. 
+
+**3. Discoverability**: Developers cannot call external contracts or integrate software libraries into applications if the former are not publicly available. By design, smart contracts are open-source; anyone can call a smart contract or fork a codebase. 
+
+## Benefits of composability {#benefits-of-composability}
+
+### Shorter development cycle
+
+Composability reduces the work that developers have to do when creating [decentralized applications](https://ethereum.org/en/dapps/#what-are-dapps) (dapps). As Naval Ravikant [puts it](https://twitter.com/naval/status/1444366754650656770): "Open source means every problem has to be solved once."
+
+Once a particular smart contract solves one problem, other developers can reuse it, so they don’t have to solve the same problem. In this way, developers can take existing software libraries and add extra functionality to create new dapps. 
+
+### Greater innovation
+
+Composability encourages innovation and experimentation. That’s because developers are free to reuse, modify, duplicate, or integrate open-source code to create desired results. 
+
+Also, development teams spend less time on basic functionality and can allocate more time to experimenting with new features. This explains why there are so many great projects in the Ethereum ecosystem today—developers can focus on improve on existing dapps and build better products. 
```

Comment:
> This explains why there are so many great projects in the Ethereum ecosystem today—developers can focus on improve on existing dapps and build better products. 

We try to avoid subjective opinions where possible.

Reply:
Understood. I'll note it for future contributions.